mig-agent/agent.go

```diff
@@ -368,6 +377,24 @@ func runAgent(runOpt runtimeOptions) (err error) {
 	// service should simply be stopped.
 	exitReason = <-ctx.Channels.Terminate
 	ctx.Channels.Log <- mig.Log{Desc: fmt.Sprintf("Shutting down agent: '%v'", exitReason)}.Emerg()
+	// send stop messags to existing persistent modules
+	if len(runningPersistentMods) > 0 {
+		for modName := range runningPersistentMods {
+			msg := modules.Message{Class: modules.MsgClassStop, Parameters: nil}
+			response := make(chan []byte)
```

Comment:
Actually the channel denoted by `response` is to be used to notify back to the caller that wants to send message to the persistent module. Inspiration was taken from here https://github.com/jvehent/galilego/blob/master/main.go#L46 . The motivation behind is that we provide a method to send messages to the persistent module but at the same time the sender can continue to work and will not block to wait for response from the module.
It's used here for a similar purpose as we send stop message to the module on its supervisor channel and wait for their answer on the `response` channel for a specific period of time. In other scenarios, it would not make sense to use the heartbeat channel or the stdout channel of the module as: Multiple senders can send their requests to the module and we are already reading them in a goroutine but we want a way to notify them in a single mapped fashion. The channel is basically used as a signalling mechanism to notify the sender. Does it makes sense ?